st2common/st2common/content/loader.py

```diff
@@ -204,6 +240,36 @@ def _get_folder(self, pack_dir, content_type):
             return None
         return path
 
+    def _get_runners_from_dir(self, base_dir):
+        result = {}
+        for runner_name in os.listdir(base_dir):
```

Comment:
We have runner alias as well. So it would be trivial to name, for instance, `mistral-v2` to `mistral_v2` and add an alias for `mistral-v2`. This also ties into [your other comment](https://github.com/StackStorm/st2/pull/2824#pullrequestreview-967994).

What type sanitation of the directory name are you thinking? Enforcing not using dash, for instance? I didn't see anything obvious in `loader.py` that adds any additional sanitation for pack loading.